src/main/java/org/opensearch/plugin/insights/core/metrics/OperationalMetricsCounter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.metrics;
+
+import java.util.Locale;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Stream;
+import org.opensearch.telemetry.metrics.Counter;
+import org.opensearch.telemetry.metrics.MetricsRegistry;
+import org.opensearch.telemetry.metrics.tags.Tags;
+
+/**
+ * Class contains all the Counters related to search query types.
+ */
+public final class OperationalMetricsCounter {
+    private static final String PREFIX = "search.insights.";
+    private static final String CLUSTER_NAME_TAG = "cluster_name";
+    private static final String UNIT = "1";
+
+    private final String clusterName;
+    private final MetricsRegistry metricsRegistry;
+    private final ConcurrentHashMap<OperationalMetric, Counter> metricCounterMap;
+
+    private static OperationalMetricsCounter instance;
+
+    /**
+     * Constructor of OperationalMetricsCounter
+     * @param metricsRegistry the OTel metrics registry
+     */
+    private OperationalMetricsCounter(String clusterName, MetricsRegistry metricsRegistry) {
+        this.clusterName = clusterName;
+        this.metricsRegistry = metricsRegistry;
+        this.metricCounterMap = new ConcurrentHashMap<>();
+        Stream.of(OperationalMetric.values()).forEach(name -> metricCounterMap.computeIfAbsent(name, this::createMetricCounter));
+    }
+
+    /**
+     * Initializes the singleton instance of OperationalMetricsCounter.
+     * This method must be called once before accessing the instance.
+     *
+     * @param clusterName the name of the cluster
+     * @param metricsRegistry the OTel metrics registry
+     */
+    public static synchronized void initialize(String clusterName, MetricsRegistry metricsRegistry) {
+        if (instance == null) {
+            instance = new OperationalMetricsCounter(clusterName, metricsRegistry);
+        }
+    }
+
+    /**
+     * Get the singleton instance of OperationalMetricsCounter.
+     *
+     * @return the singleton instance
+     * @throws IllegalStateException if the instance is not yet initialized
+     */
+    public static synchronized OperationalMetricsCounter getInstance() {
+        if (instance == null) {
+            throw new IllegalStateException("OperationalMetricsCounter is not initialized. Call initialize() first.");
+        }
+        return instance;
+    }
+
+    /**
+     * Increment the operational metrics counter, attaching necessary tags
+     * @param metricName name of the metric
+     */
+    public void incrementCounter(OperationalMetric metricName) {
```

Comment:
hmm, I don't think it's a good idea, since it would be anti-pattern - In object-oriented design, **behavior should be encapsulated within objects, and Static methods shouldn't rely on `instances`**, since this class follows the singleton pattern, we should have one and only one instance of this class. If we don't want to even use the singleton pattern and make the whole class static, managing state like the `clusterName` (and potentially other state in the future) becomes complicated because there would be no clear place to store that data. You'd have to pass more arguments to the method, which will then increase the complexity of the argument we passes/stored in all services like top n and categorizer.

Also, Static methods are harder to test because you can't easily mock or substitute them in unit tests without frameworks like PowerMockito. Keeping methods as instance methods allows you to mock the instance for better testability and maintainability.